diff --git a/builder/cc_builder.rs b/builder/cc_builder.rs
index 6ed6915..a0910cc 100644
--- a/builder/cc_builder.rs
+++ b/builder/cc_builder.rs
@@ -701,7 +701,9 @@ impl CcBuilder {
                 cc_build.define("AWS_LC_URANDOM_NEEDS_U32", Some("1"));
             }
         }
-        self.memcmp_check();
+        if !env_var_to_bool("AWS_LC_SYS_SKIP_MEMCMP_CHECK").unwrap_or(false) {
+            self.memcmp_check();
+        }
     }
 }
