diff --git a/build.rs b/build.rs
index 0000000..1111111 100644
--- a/build.rs	2021-02-16 01:14:27.000000000 -0600
+++ b/build.rs	2026-01-27 21:06:27.047724269 -0600
@@ -56,6 +56,16 @@
         println!("cargo:rustc-flags={}", x);
     }
 
+    if std::env::var("NCURSES_RS_SKIP_PROBE").is_ok() {
+        if std::env::var("CARGO_CFG_TARGET_POINTER_WIDTH").as_deref() == Ok("64") {
+            println!("cargo:rustc-cfg=feature=\"wide_chtype\"");
+        }
+        if std::env::var("NCURSES_RS_MOUSE_V1").as_deref() == Ok("1") {
+            println!("cargo:rustc-cfg=feature=\"mouse_v1\"");
+        }
+        return;
+    }
+
     check_chtype_size(&ncurses_lib);
 }
 
