diff -uNr rust/platform/triple_mappings.bzl rust/platform/triple_mappings.bzl
--- a/rust/platform/triple_mappings.bzl
+++ b/rust/platform/triple_mappings.bzl
@@ -24,6 +25,7 @@
 SUPPORTED_T1_PLATFORM_TRIPLES = {
     "aarch64-apple-darwin": _support(std = True, host_tools = True),
     "aarch64-unknown-linux-gnu": _support(std = True, host_tools = True),
+    "aarch64-unknown-linux-musl": _support(std = True, host_tools = True),
     "aarch64-unknown-nixos-gnu": _support(std = True, host_tools = True),  # Same as `aarch64-unknown-linux-gnu` but with `@platforms//os:nixos`.
     "i686-apple-darwin": _support(std = True, host_tools = True),
     "i686-pc-windows-msvc": _support(std = True, host_tools = True),
