diff --git a/webrtc-audio-processing/webrtc/rtc_base/logging.cc b/webrtc-audio-processing/webrtc/rtc_base/logging.cc
--- a/webrtc-audio-processing/webrtc/rtc_base/logging.cc
+++ b/webrtc-audio-processing/webrtc/rtc_base/logging.cc
@@ -24,6 +24,6 @@
 #undef ERROR  // wingdi.h
 #endif
 
 #if defined(WEBRTC_MAC) && !defined(WEBRTC_IOS)
-#include <CoreServices/CoreServices.h>
+#include <CoreFoundation/CoreFoundation.h>
 #elif defined(WEBRTC_ANDROID)
 #include <android/log.h>
